### DISCUSSION TITLE
Vantagens de usar o Hyper.App

### DISCUSSION THREAD

**Discussion by expalmer** [frontendbr/forum#332] [Perguntas e Respostas]
Olá pessoal, já viram o projeto [Hyper.app](https://hyper.is/) certo?

Gostaria de saber o que acharam e quais as vantagens de uso em vez dos nosso tradicionais terminais.

**Comment by mtmr0x** [comment]
Aqui estão todas as vantagens: https://lh6.ggpht.com/p_4CpEbK-JA2sx_Kq-CVNTqQeiZADrMJKsol-GV8QK8aJ6gCXgTVJ98SdUCmRyUEJUg=w300

**Comment by mtmr0x** [comment]
Ok, brincadeiras a parte, o que acontece é o seguinte: é uma interface que emula o terminal. É feito em JS, e roda um tantinho mais pesado do que o terminal nativo, sem nenhuma vantagem. 🙄 

Se quiser um terminal com reais power ups, vai de iTerm (caso use Mac) e seja feliz. O iTerm faz um bom trampo e é super leve, bem comparável ao próprio terminal em Unix OS.

**Comment by bukinoshita** [comment]
@matmarsiglio 

> É feito em JS, e roda um tantinho mais pesado do que o terminal nativo

Benchmark?

**Comment by mtmr0x** [comment]
@BuKinoshita https://www.dicio.com.br/tantinho/ Terminal é uma interface[0], HyperApp é uma interface[1] que manda coisa pra interface[0], pelo menos mais 1 processo. Desconsiderável? Bem provável. Mas em terminologia, um tantinho mais pesado, sim. :)

e o ponto é: nenhuma vantagem. A não ser que "boniteza" escreva melhor `gulp watch` hahaha

**Comment by expalmer** [comment]
pois é... eu uso iTerm já a anos, e achei o hyper bem bonitinho, mas só...por isso a dúvida se ele tinha alguma feature que fosse atrativa e útil...mas....

**Comment by paoloalmeida** [comment]
Até onde eu acompanhei, tinha uma issue aberta que não rolava usar o **~**, nem as **""**... Ai já matou 50% do meu flow kkk desencanei dele também. Mas é bonito, e só.

**Comment by fernahh** [comment]
@matmarsiglio @BuKinoshita sobre benchmarks:

<img width="602" alt="90e31942-537c-11e6-9f38-8ef07834d481" src="https://cloud.githubusercontent.com/assets/2369851/19542077/40428412-964a-11e6-8287-5d3c235b3b9c.png">

> https://github.com/zeit/hyper/issues/296#issuecomment-235460449

**Comment by expalmer** [comment]
@matmarsiglio caraca rapaz...pô o Fedor Indutny falando isso...

**Comment by RuanAragao** [comment]
Nunca usei, mas tenho quase certeza que é uma parada só para comer mais memória. Ando com trauma dessas coisas.

**Comment by mtmr0x** [comment]
@fernahh oloco, aí sim! E é, o iTerm eu já o sentia pesado sempre, mas não sabia que o HyperTerm tinha vantagem de performance em cima dele dessa forma. Massa, só eles corrigirem umas cagadas pra galera mais hard user de terminal e dá pra considerar brincar 😄 . Por hora o bom e velho Terminal continua meu melhor amigo.

**Comment by felipecodes** [comment]
Tem uma desvantagem: não roda `imgcat` 🙁

**Comment by expalmer** [comment]
Senhores, por acaso alguém sabe como usar um Blur no Hyper ? Alguém tem alguma configuração bacana e queira compartilhar...
Estou achando muito bom o Hyper.

**Comment by felipefialho** [comment]
Vou começar a usar também.

**Comment by diogomoretti** [comment]
Eu cheguei a usar assim que saiu e também dei outra chance agora na 1.0 =] O fato é que é meio lento e ainda com vários bugs. Tive problema para colar URLs... enfim, ele é até bonitinho e tals, mas assim como a maioria dos comentários aqui, não pretendo trocar o meu iTerm tão cedo!

**Comment by felipefialho** [comment]
Bom to usando desde ontem.

Até agora nada a reclamar, funciona mais rápido (bem mais) do que o iTerm e é legal usar CSS e JavaScript para configurar as coisas.

Não encontrei nenhum bug ainda. E o maior problema que vi, foi não ter opção de toggle, mas contornei facilmente com um script.

Recomendo.

**Comment by Adejair** [comment]
É um terminal muito bom por causa das personalizações, você personaliza a sua forma e seu jeito, assim fica melhor até para você automatizar seu workflow.

A um tempo atrás tive um problema com ele em relação a acentuação e outras coisinhas. ([https://github.com/zeit/hyper/issues/66](problema))

Mais foi corrigido, com demora mais foi, mais é isso irmão, é super bacana de você utilizar se você quiser personalizar seu workflow da sua maneira, inclusive, vou aproveitar e deixar um merchan do [tema ](https://github.com/Adejair/dracula-hyperterm)que eu fiz na época aqui, usem ele HAHA.

**Comment by kazzkiq** [comment]
O bixinho tá sofrendo na minha mão desde o inicio da semana.

<img width="625" alt="captura de tela 2016-12-22 as 15 03 21" src="https://cloud.githubusercontent.com/assets/1953194/21433607/9e6258bc-c857-11e6-9044-acba7186b33f.png">


Em termos de performance, nada a reclamar. Nem sinto diferença de velocidade pro app de terminal padrão do Mac.

O único bug que encontrei nesse meio tempo foi dando resize na janela. As vezes um texto longo não se adequa quando você redimensiona o Hyper, fora isso, não vi mais nenhum bug.

Se continuar se comportando bem assim acho que toma o posto do Terminal padrão do Mac. :1st_place_medal:

**Comment by brunoqueiros** [comment]
Estou gostando também, achei apenas alguns pequenos bugs mas nada que atrapalhe tanto.

Não funciona o atalho do bash para `reverse-i-search`
[https://github.com/zeit/hyper/issues/1178](https://github.com/zeit/hyper/issues/1178)

E o scroll com o mouse também não funciona quando o log do `git diff` é muito longo, tenho que ficar utilizando o teclado para mover o cursor.

**Comment by renansalvino** [comment]
Olá povo, alguém aqui sabe como habilitar emojis no terminal Hyper? Assim como o tema CodingPhase: https://github.com/codingphasedotcom/CodingPhaseThemes gostaria de habilitar isso no hyper, mas pelo que eu encontrei tem apenas no Iterm do Mac e eu uso uma distribuição Linux Kubuntu.
